src/crystal/lib_iconv.cr

```diff
@@ -0,0 +1,18 @@
+require "c/stddef"
+
+{% if flag?(:without_iconv) %}
+  {% raise "The `without_iconv` flag is preventing you to use the LibIconv module" %}
+{% end %}
+
+@[Link("iconv")]
```

Comment:
Hmm, does this mean everyone that wants to use strings on windows need to have the iconv lib installed? GNU libiconv is LGPL, which allows *dynamic* linking, but not static.

http://www.gnu.org/licenses/gpl-faq.html#LGPLStaticVsDynamic

Reply:
It turns out MPIR has the same issue. Same goes for GMP on non-Windows platforms actually.

Reply:
It's not "everyone who want's to use string". If you don't need the encoding support, you can pass `-Dwithout_iconv`. But I guess that doesn't help much.

Reply:
I think it is fine as long as the compiler itself can be compiled without those LGPL components on Windows (which is already the case), but we have to be very clear about all the third-party software used by Crystal and which standard library types link to them, probably directly under the Crystal book's home.

Reply:
Everyone that wants to use strings on Windows will have to provide source code.

This is not a new problem or a problem for this pull request to solve.

Actually maybe it's not even a problem at all. Linking against dynamic libraries works just as well (exact same method, even), just that nobody bothered to set it up that way yet, and the CI builds happen to ship static libraries.

Reply:
Let's continue the discussion in https://github.com/crystal-lang/crystal/issues/11488